client/two-column-tasks/index.js

```diff
@@ -0,0 +1,286 @@
+		activePlugins,
+		countryCode,
+		freeExtensions,
+		installedPlugins,
+		isJetpackConnected,
+		onboardingStatus,
+		profileItems,
+		isSetupTaskListHidden,
+		dismissedTasks,
+		remindMeLaterTasks,
+		isTaskListComplete,
+		hasCompleteAddress,
+		trackedCompletedActions,
+		isResolving,
+	} = useSelect( taskDashboardSelect );
+
+	const [ isCartModalOpen, setIsCartModalOpen ] = useState( false );
+	const [ isLoadingExperiment, experimentAssignment ] = useExperiment(
+		'woocommerce_tasklist_progression'
+	);
+
+	useEffect( () => {
+		document.body.classList.add( 'woocommerce-onboarding' );
+		document.body.classList.add( 'woocommerce-task-dashboard__body' );
+	}, [] );
+
+	const getTaskStartedCount = ( taskName ) => {
+		const trackedStartedTasks =
+			userPreferences.task_list_tracked_started_tasks;
+		if ( ! trackedStartedTasks || ! trackedStartedTasks[ taskName ] ) {
+			return 0;
+		}
+		return trackedStartedTasks[ taskName ];
+	};
+
+	const updateTrackStartedCount = ( taskName, newCount ) => {
+		const trackedStartedTasks =
+			userPreferences.task_list_tracked_started_tasks || {};
+		userPreferences.updateUserPreferences( {
+			task_list_tracked_started_tasks: {
+				...( trackedStartedTasks || {} ),
+				[ taskName ]: newCount,
+			},
+		} );
+	};
+
+	const isTaskCompleted = ( taskName ) => {
+		if ( ! trackedCompletedTasks ) {
+			return false;
+		}
+		return trackedCompletedTasks.includes( taskName );
+	};
+
+	const onTaskSelect = ( taskName ) => {
+		const trackStartedCount = getTaskStartedCount( taskName );
+		recordEvent( 'tasklist_click', {
+			task_name: taskName,
+		} );
+
+		if ( ! isTaskCompleted( taskName ) ) {
+			updateTrackStartedCount( taskName, trackStartedCount + 1 );
+		}
+	};
+
+	const toggleCartModal = () => {
+		if ( ! isCartModalOpen ) {
+			recordEvent( 'tasklist_purchase_extensions' );
+		}
+
+		setIsCartModalOpen( ! isCartModalOpen );
+	};
+
+	const getCurrentTask = ( tasks ) => {
+		const { task } = query;
+		const currentTask = tasks.find( ( s ) => s.key === task );
+
+		if ( ! currentTask ) {
+			return null;
+		}
+
+		return currentTask;
+	};
+
+	const allTasks = getAllTasks( {
+		activePlugins,
+		countryCode,
+		createNotice,
+		freeExtensions,
+		installAndActivatePlugins,
+		installedPlugins,
+		isJetpackConnected,
+		onboardingStatus,
+		profileItems,
+		query,
+		toggleCartModal,
+		onTaskSelect,
+		hasCompleteAddress,
+		trackedCompletedActions,
+	} );
+
+	const { extension, setup: setupTasks } = allTasks;
+	const { task } = query;
+
+	const extensionTasks =
+		Array.isArray( extension ) && extension.sort( taskSort );
+
+	const currentTask = getCurrentTask( [
+		...( extensionTasks || [] ),
+		...( setupTasks || [] ),
+	] );
+
+	if ( task && ! currentTask ) {
+		return null;
+	}
+	if ( currentTask ) {
+		return (
+			<TaskStep taskContainer={ currentTask.container } query={ query } />
+		);
+	}
+
+	if ( isResolving ) {
+		return <TaskListPlaceholder />;
+	}
+
+	return (
+		<>
+			{ setupTasks &&
```

Comment:
Nitpick: Instead of using ternary here, I would love it if we can move the logic and <TaskListPlaceholder> component to its own block, similar to the block in lines [240 - 242](https://github.com/woocommerce/woocommerce-admin/pull/7669/files#diff-529315666a7e750da7646157bf147160a7eaae0568474819fb21be8a8dc78916R240-R242). This should make it easier on the eyes for the last return of the component.

Reply:
Thank you @ilyasfoo I agree with that, but the code is from https://github.com/woocommerce/woocommerce-admin/blob/main/client/task-list/task-list.js. I would like to minimize changing the existing code in case we want to make this experiment permanent later on. It would be easier to copy over any changes added to the existing task list.

1. The existing test should work with the two-column task list. I'll test it out.
2. Yup! I think I'll need to go over styling again.